herbert_core/utilities/geometry/box_intersect.m

```diff
@@ -0,0 +1,156 @@
+function inter_points = box_intersect(box_minmax,cross_plain)
+% Calculate intersection points between the box in ND (2,3,4)
+% and line/plain/hyperplane of dimension (N-1)D
+%
+% Inputs:
+% box_minmax -- NDx2 array of min and max points of the box, to intersect with.
+%
+% cross_plain -- NDxND array of points defining plain in the appropriate
+%                dimensions. The coordinates defined along the first
+%                dimension and the second dimention correspont to number of
+%                points, the plain is passing through (has to be ND). If
+%                there are ND-1 points, the missing point assumed to be
+%                equal to 0.
+% Outputs:
+% inter_points - NDxNp where NP -- the number of intesection points
+%                array of points, defining intersection between the
+%                edges of the box and the line/plain/hyperplain defined as
+%                the second argument
+%                If no intersection points are present, the array is empty
```

Comment:
So if the box is a square (2D box) and the cross-plane is a line, then there are up to 2 intersection points with the box. If the box is a cube (3D box) there are up to 6 intersection points. It is a total guess on my part that there are 24 intersections in 4D. But enumerating this to check that I am correct about what what your're doing. Are the intersection points (xyzt) coordinate tuples or some other representation?

Reply:
It can be up to 2 intesection points in 2D, up to 4 in 3D and ? on 4D (is it 8 or 6? or something else...) But the plain passing through the edges produces degenerated points (ND-level degeneracy). So the arrays are pre-allocated with excess size.

Reply:
not sure what exactly you want here. I've added couple examples to comments, though I, if see it, would not trust them and go to unit tests to check.